website/views.py

```diff
@@ -3348,6 +3350,18 @@ def get(self, request, *args, **kwargs):
         return render(request, self.template_name, context)
 
 
+def is_valid_url(url):
+    try:
+        parsed_url = urlparse(url)
+        if parsed_url.scheme not in ["http", "https"]:
+            return False
+        if re.match(r"^(localhost|127\.|10\.|172\.16\.|192\.168\.)", parsed_url.hostname):
```

Comment:
This looks like an attempt to exclude requests to private network IP addresses. In this case the regex is incomplete, e.g. for `172.16.0.0/12` it won't catch `172.17.*` IPs and so on.